dataset_reader/tar_compound_reader.py

```diff
@@ -0,0 +1,54 @@
+import json
+import tarfile
+from io import BytesIO
+from typing import Iterator
+
+import numpy as np
+
+from dataset_reader.base_reader import BaseReader, Query, Record
+
+
+class TarCompoundReader(BaseReader):
```

Comment:
Yeah, I agree it might be a good idea to reuse some parts, however, that requires some refactoring of the `JsonReader`, as it is dependent on some constants. I'll do that.